app/actions/GovernanceActions.js

```diff
@@ -153,41 +153,47 @@ export const compareInventory = () => async (dispatch, getState) => {
   try {
     const oldInventory = sel.inventory(getState());
     const piURL = sel.politeiaURL(getState());
-    let oldProposals = sel.proposals(getState());
-    const keys = Object.keys(oldInventory);
+    const oldProposals = sel.proposals(getState());
+    const newProposalsList = getDefaultInventory();
 
     const { data } = await pi.getTokenInventory(piURL);
     const inventory = updateInventoryFromApiData(data);
 
     let isDifferent = false;
 
     // Check if old proposals list have had some proposal removed from inventory and
-    // also remove it from the list.
-    for (let i = 0; i < keys.length; i++) {
-      const key = keys[i];
-      oldProposals[key].forEach( (proposal, index) => {
-        if (!inventory[key].includes(proposal.token)) {
-          oldProposals[key].splice(index, 1);
+    // also remove it from proposal's list.
+    Object.keys(oldProposals).forEach( key => {
+      newProposalsList[key] = oldProposals[key].reduce((acc, p) => {
+        if(inventory[key].includes(p.token)) {
+          acc.push(p);
+        } else {
+          // Remove token from old inventory so we can re-fetch it.
+          oldInventory[key].splice(oldInventory[key].indexOf(p.token), 1)
```

Comment:
I still think modifying `oldInventory`, even if it doesn't produce local side-effects in this function can cause side-effects in the app. 

As a rule of thumb you should **not** modify any objects returned from the redux state storage (even if they were indirectly returned via a select). I'd suggest duplicating `oldInventory` before starting the main `Object.keys(oldProposals).forEach(...)` loop.

Reply:
Making a copy of oldInventory now